plugins/modules/scc_product.py

```diff
@@ -63,7 +63,9 @@
 def main():
     module = KatelloAnsibleModule(
         foreman_spec=dict(
-            scc_product=dict(required=True, type='entity', aliases=['friendly_name'], scope=['scc_account'], thin=False),
+            # scc_product search by friendly_name alias is deprecated in version v1.8.6
+            # scc_product=dict(required=True, type='entity', aliases=['friendly_name'], scope=['scc_account'], thin=False),
+            scc_product=dict(required=True, type='entity', scope=['scc_account'], thin=False),
```

Comment:
If I understand correctly, I don't think we actually need to remove this alias. (If we do remove it we also need to change the example in the module documentation). This alias just means users can use either `scc_product` or `friendly_name`, to specify the search string for the module. Whether that search string then looks over the `name` field, or the `friendly_name` field from the API replies is controlled exclusively via the `ENTITY_KEYS` dict, no?

Of course it might be a bit misleading to leave the alias `friendly_name`, and not actually have it search over the `friendly_name` fields from the API replies, but my intuition is that this is less bad than breaking peoples plays.

Reply:
If the new "unique name" looks different than what was the "unique friendly_name", users playbooks will break anyway. In that case removing the alias might make the need and the knowlege where to adjust more obvious.

Reply:
I guess this would become easier to evaluate once some new test fixtures are added to this PR (I presume we will need new test fixtures for this change :wink:).

Reply:
I think I am in favor of chucking out the `friendly_name` Alias. After all, `name` was never the same thing as `friendly_name`, so in some sense it never made sense as an Alias.